lib/JSON/Schema/Fit.pm

```diff
@@ -174,11 +175,20 @@ sub get_adjusted {
 
     $struc = _default($schema) if !defined $struc && $self->fill_defaults;
 
+    if ($self->fill_defaults && exists($schema->{enum}) && exists($schema->{default})) {
```

Comment:
There should be a separate attribute like replace_invalids, disabled by default.
And check here should be refactored, it may be extended later.

if ($self->replace_invalids && exists($schema->{default}) && $self->_is_invalid($struc, $schema)) {
  $struc = _default($schema);
}

Reply:
Sure. Will also check how Data::Compare behaves when comparing numbers with strings (will add a couple of tests for that too)